File: src/main/java/com/saybetter/domain/solution/application/impl/SolutionService.java
package com.saybetter.domain.solution.application.impl;

import org.springframework.stereotype.Service;

import com.saybetter.domain.solution.dao.repository.SolutionReadRepository;
import com.saybetter.domain.solution.dao.repository.SolutionWriteRepository;
import com.saybetter.domain.solution.domain.Solution;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@Slf4j
@Service
@RequiredArgsConstructor
public class SolutionService {

	private final SolutionReadRepository solutionReadRepository;
	private final SolutionWriteRepository solutionWriteRepository;

	public Solution createSolution(Solution newSolution) {
		return solutionWriteRepository.save(newSolution);
	}
}

File: core/core-api/src/main/kotlin/io/say/better/domain/solution/application/impl/SolutionService.kt
package io.say.better.domain.solution.application.impl

import io.say.better.core.common.code.status.ErrorStatus.START_SOLUTION_NOT_FOUND
import io.say.better.core.common.exception.GeneralException
import io.say.better.storage.mysql.domains.solution.entity.Solution
import io.say.better.storage.mysql.domains.solution.repository.SolutionReadRepository
import io.say.better.storage.mysql.domains.solution.repository.SolutionWriteRepository
import org.springframework.stereotype.Service

@Service
class SolutionService(
    private val solutionReadRepository: SolutionReadRepository,
    private val solutionWriteRepository: SolutionWriteRepository,
) {
    fun createSolution(newSolution: Solution): Solution = solutionWriteRepository.save(newSolution)

    fun getSolution(solutionId: Long): Solution =
        solutionReadRepository.findById(solutionId).orElseThrow {
            throw GeneralException(START_SOLUTION_NOT_FOUND)
        }
}
